cmc-ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/PartyMapper.java

```diff
@@ -0,0 +1,116 @@
+
+    @Autowired
+    public PartyMapper(IndividualMapper individualMapper, CompanyMapper companyMapper,
+                       OrganisationMapper organisationMapper, SoleTraderMapper soleTraderMapper,
+                       AddressMapper addressMapper, RepresentativeMapper representativeMapper) {
+        this.individualMapper = individualMapper;
+        this.companyMapper = companyMapper;
+        this.organisationMapper = organisationMapper;
+        this.soleTraderMapper = soleTraderMapper;
+        this.addressMapper = addressMapper;
+        this.representativeMapper = representativeMapper;
+    }
+
+    @Override
+    public CCDParty to(Party party) {
+        final CCDParty.CCDPartyBuilder builder = CCDParty.builder();
+
+        if (party instanceof Individual) {
+            builder.type(CCDPartyType.INDIVIDUAL);
+            Individual individual = (Individual) party;
+            builder.individual(individualMapper.to(individual));
+        } else if (party instanceof Company) {
+            builder.type(CCDPartyType.COMPANY);
+            Company company = (Company) party;
+            builder.company(companyMapper.to(company));
+        } else if (party instanceof Organisation) {
+            builder.type(CCDPartyType.ORGANISATION);
+            Organisation organisation = (Organisation) party;
+            builder.organisation(organisationMapper.to(organisation));
+        } else if (party instanceof SoleTrader) {
+            builder.type(CCDPartyType.SOLE_TRADER);
+            SoleTrader soleTrader = (SoleTrader) party;
+            builder.soleTrader(soleTraderMapper.to(soleTrader));
+        }
+
+        return builder.build();
+    }
+
+    @Override
+    public Party from(CCDParty ccdParty) {
+        switch (ccdParty.getType()) {
+            case COMPANY:
+                final CCDCompany ccdCompany = ccdParty.getCompany();
+                return new Company(ccdCompany.getName(),
+                    addressMapper.from(ccdCompany.getAddress()),
+                    addressMapper.from(ccdCompany.getCorrespondenceAddress()),
+                    ccdCompany.getMobilePhone(),
+                    representativeMapper.from(ccdCompany.getRepresentative()),
+                    ccdCompany.getContactPerson());
+            case INDIVIDUAL:
+                final CCDIndividual ccdIndividual = ccdParty.getIndividual();
+                return new Individual(
+                    ccdIndividual.getName(),
+                    addressMapper.from(ccdIndividual.getAddress()),
+                    addressMapper.from(ccdIndividual.getCorrespondenceAddress()),
+                    ccdIndividual.getMobilePhone(),
+                    representativeMapper.from(ccdIndividual.getRepresentative()),
+                    ccdIndividual.getTitle(),
+                    LocalDate.parse(ccdIndividual.getDateOfBirth(), DateTimeFormatter.ISO_DATE)
+                );
+            case SOLE_TRADER:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 10 to at most 5, for example by extracting code into methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS1151)